web_widget_numeric_step/static/src/js/numeric_step.js

```diff
@@ -0,0 +1,239 @@
+    };
+
+    var NumericStep = FieldFloat.extend({
+        template: 'web_widget_numeric_step',
+        className: 'o_field_numeric_step o_field_number',
+        events: _.extend({}, FieldFloat.prototype.events, {
+            'mousedown .btn_numeric_step': '_onStepMouseDown',
+            'touchstart .btn_numeric_step': '_onStepMouseDown',
+            'click .btn_numeric_step': '_onStepClick',
+            'wheel .input_numeric_step': '_onWheel',
+            'keydown .input_numeric_step': '_onKeyDown',
+            'change .input_numeric_step': '_onChange',
+        }),
+        supportedFieldTypes: ['float', 'integer'],
+
+        // Values used to mouse down smooth speed feature
+        DEF_CLICK_DELAY: 400,
+        MIN_DELAY: 50,
+        SUBSTRACT_DELAY_STEP: 25,
+
+        /**
+         * Add global events listeners
+         *
+         * @override
+         */
+        start: function () {
+            var self = this;
+            this._click_delay = this.DEF_CLICK_DELAY;
+            this._autoStep = false;
+            return this._super.apply(this, arguments).then(function () {
+                document.addEventListener(
+                    'mouseup', $.proxy(self._onMouseUp, self), false);
+                document.addEventListener(
+                    'touchend', $.proxy(self._onMouseUp, self), false);
+            });
+        },
+
+        /**
+         * Transform Odoo to Input
+         *
+         * @override
+         */
+        _formatValue: function (value) {
+            if (this.mode === 'edit' && this.nodeOptions.type === 'tel') {
+                return this._sanitizeNumberValue(value);
+            }
+            return this._super.apply(this, arguments);
+        },
+
+        /**
+         * Transform Input to Odoo
+         *
+         * @override
+         */
+        _parseValue: function () {
+            var parsedVal = this._super.apply(this, arguments);
+            if (this.mode === 'edit' && this.nodeOptions.type === 'tel') {
+                return Number(parsedVal) || 0;
+            }
+            return parsedVal;
+        },
+
+        /**
+         * Adds HTML attributes to the input
+         *
+         * @override
+         */
+        _prepareInput: function () {
+            var result = this._super.apply(this, arguments);
+            this.$input.attr(_.pick(this.nodeOptions, ['placeholder', 'type']));
+            return result;
+        },
+
+        /**
+         * The widget uses 'tel' type because offers a good layout on mobiles
+         * and can accept alphanumeric characters.
+         * The bad news is that require implement all good 'number' type
+         * features like the minus and plus buttons, steps, min and max...
+         * Perhaps in a near future this can be improved to have the best of
+         * two types without hacky developments.
+         *
+         * @override
+         */
+        _render: function () {
+            this.nodeOptions.type = "tel";
```

Comment:
Isn't this done already in the template? :thinking:

Reply:
The 'type' in the template really does nothing... 'InputField' hard-write the type.